src/test/fs_tests.cpp

```diff
@@ -171,8 +171,14 @@ BOOST_AUTO_TEST_CASE(create_directories)
     BOOST_CHECK(fs::is_directory(symlink));
     BOOST_CHECK(!fs::create_directories(symlink));
 
-    fs::remove(symlink);
     fs::remove(dir);
+    BOOST_CHECK(!fs::exists(symlink));
+    BOOST_CHECK(fs::is_symlink(symlink));
+    BOOST_CHECK(!fs::is_directory(symlink));
+    // Next check is disabled due to a bug in libc++-11
+    // See https://github.com/bitcoin/bitcoin/pull/24432#issuecomment-1049060318
+    // BOOST_CHECK_THROW(fs::create_directories(symlink), fs::filesystem_error);
```

Comment:
for reference, the bug still exists and can be reproduced on (for example) Ubuntu Jammy clang-11:

```
# clang-11 --version 
Ubuntu clang version 11.1.0-6
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

# ./src/test/test_bitcoin -t fs_tests
Running 5 test cases...
test/fs_tests.cpp(180): error: in "fs_tests/create_directories": exception fs::filesystem_error expected but not raised

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"